File: src/main/java/org/checkerframework/checker/mungo/typestate/graph/exceptions/DuplicateState.java
package org.checkerframework.checker.mungo.typestate.graph.exceptions;

import org.checkerframework.checker.mungo.typestate.ast.TStateNode;

public class DuplicateState extends TypestateError {

  public static final long serialVersionUID = 0L;

  public final TStateNode first, second;

  public DuplicateState(TStateNode first, TStateNode second) {
    this.first = first;
    this.second = second;
  }

  @Override
  public String getMessage() {
    return String.format("Duplicate state %s in %s at %s and %s", first.name, first.pos.getBasename(), first.pos.getLineCol(), second.pos.getLineCol());
  }
}

File: src/main/kotlin/org/checkerframework/checker/mungo/typestate/graph/exceptions/DuplicateState.kt
package org.checkerframework.checker.mungo.typestate.graph.exceptions

import org.checkerframework.checker.mungo.typestate.ast.TStateNode

class DuplicateState(val first: TStateNode, val second: TStateNode) : TypestateError() {
  override val message: String
    get() = String.format("Duplicate state %s in %s at %s and %s", first.name, first.pos.basename, first.pos.lineCol, second.pos.lineCol)

  companion object {
    const val serialVersionUID = 0L
  }
}
